cli/cmdlineparser.cpp

```diff
@@ -1616,13 +1628,15 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
     return Result::Success;
 }
 
-void CmdLineParser::printHelp() const
+void CmdLineParser::printHelp(bool premium) const
 {
-    const std::string manualUrl(isCppcheckPremium() ?
-                                "https://cppcheck.sourceforge.io/manual.pdf" :
-                                "https://files.cppchecksolutions.com/manual.pdf");
+    // TODO: fetch URL from config like product name?
+    const std::string manualUrl(premium ?
+                                "https://files.cppchecksolutions.com/manual.pdf" :
+                                "https://cppcheck.sourceforge.io/manual.pdf");
```

Comment:
These were switched.

Reply:
Also we should not have the custom URL hardcoded and read it from `cppcheck.cfg` like the product name instead. I files https://trac.cppcheck.net/ticket/14208 about this.